src/index.js

```diff
@@ -269,21 +422,18 @@ class ClientAuthOIDC {
    *
    * @param [providerUri] {string} If the provider URI is already known to the
    *   app developer, just pass it through, no need to take further action.
-   * @return {Promise<string>}
+   *
+   * @return {string|null}
    */
   selectProvider (providerUri) {
-    if (providerUri) {
-      return Promise.resolve(providerUri)
-    }
+    providerUri = providerUri || this.currentProvider()
 
-    // Attempt to find a provider based on the 'state' param of the current URI
-    providerUri = this.providerFromCurrentUri()
     if (providerUri) {
-      return Promise.resolve(providerUri)
+      return providerUri
     }
 
-    // Lastly, kick off a Select Provider popup window workflow
-    return this.providerFromUI()
+    // If not available, kick off a Select Provider popup window workflow
+    this.selectProviderUI()
```

Comment:
That was my initial thought too, when I thought there was a way that this function can actually return the provider uri. Currently, though, the `selectProviderUI()` function just opens the Select Provider popup window, and sets up the event listeners on it.
What would you like to see as a return value - the handle to the opened popup window?